.github/workflows/deploy.yml

```diff
@@ -76,17 +81,26 @@ jobs:
 
       ### SET TARGET_DIR environment variable for dev version
       - name: Set TARGET_DIR for dev
-        if: ${{ (github.repository_owner == env.REPO_OWNER) && (github.ref_type == 'branch') && (github.event.base_ref == 'develop') }}
+        if: github.repository_owner == env.REPO_OWNER && github.ref == 'refs/heads/develop'
         run: echo "TARGET_DIR=dev" >> $GITHUB_ENV
 
       ### SET TARGET_DIR environment variable for tagged release
       - name: Set TARGET_DIR for tagged release
-        if: ${{ (github.repository_owner == env.REPO_OWNER) && (github.ref_type == 'tag')  && (github.event.base_ref == 'stable') }}
+        if: github.repository_owner == env.REPO_OWNER  && startsWith(github.ref, 'refs/tags/')
         run: echo "TARGET_DIR=${{ github.ref_name }}" >> $GITHUB_ENV
 
+      ### VALIDATE TARGET NAME
+      - name: Validate TARGET_DIR
+        run: echo "TARGET_DIR=$TARGET_DIR"
+
+      ### Exit if ref is not develop branch or release tag
```

Comment:
This could probably be moved up the chain, since it doesn't require anything else to run (I don't think?). There's no point getting this far if we're just going to exit...

Reply:
We wanted to run the build and artifact upload before exiting (on each branch that triggers the workflow). But you are right, we could probably exit before setting the target_dir variable.